src/Agent.Listener/Environment.cs

```diff
@@ -0,0 +1,338 @@
+using Microsoft.VisualStudio.Services.Agent.Util;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.Services.Agent.Listener
+{
+    [ServiceLocator(Default = typeof(Environment))]
+    public interface IEnvironment : IAgentService
+    {
+        Task<Dictionary<string, string>> GetCapabilities(string agentName, CancellationToken token);
```

Comment:
I believe async is not valid for functions in interfaces. I get a compilation error: "The modifier 'async' is not valid for this item"